### PR TITLE
Add support for file ignoring

### LINKED ISSUE #87: Exclude certain files from formatting
I'm not able to use the plugin in bigger projects like [esl/MongooseIM](https://github.com/esl/MongooseIM) or [esl/escalus](https://github.com/esl/escalus) because of some files crashing the plugin. The crash is most probably due to macro expansion (which was already mentioned in issues of this project). It'd helpful to exclude some files from formatting, this is probably not possible currently with the `files` option which takes a `wildcard` to get list of files to format.

### PR DESCRIPTION
**Pull request by juanbono**
Fixes #87 
- [x] Add attribute-based file ignoring (add the attribute `-format(ignore)` to a file for that)
    - [x] Add tests
    - [x] Update documentation
- [x] Add config based file ignoring (add a tuple `{ignore, [Wildcard::file:filename_all()]}` to the formatter config in `rebar.config`)
    - [x] Add tests
    - [x] Update documentation

### CODE REVIEW

File: test_app/src/ignored_file.erl [hunk 1]
```diff
@@ -0,0 +1,9 @@
+-module(ignored_file).
+
+-format(#{ignore => true}).
```
[the remark this answers is not in the document]
Comment (rjcoelho) [reply test_app/src/ignored_file.erl:3]: I prefer -indent(Options) where Options is the same map you give on rebar.config

[the remark this answers is not in the document]
Comment (elbrujohalcon) [reply test_app/src/ignored_file.erl:3]: But… if you use `ignore`… everything else is meaningless.

[the remark this answers is not in the document]
Comment (elbrujohalcon) [reply test_app/src/ignored_file.erl:3]: And… `-format #{… => …}.` already exists, by the way

File: src/rebar3_formatter.erl [hunk 2]
```diff
@@ -18,27 +18,33 @@
...
-      NewFile ->
-          case get_quick_ast(NewFile) of
-            QuickAST ->
-                Result;
-            _ ->
-                erlang:error({modified_ast, File, NewFile})
-          end
+    case apply_per_file_opts(File, Opts) of
+        ignore ->
+            unchanged;
```
Comment (elbrujohalcon) [inline src/rebar3_formatter.erl:24]: We might consider (maybe in a separate ticket) to change this to `ignored`.

Comment (elbrujohalcon) [reply src/rebar3_formatter.erl:24]: Actually… shouldn't we write the file, but unchanged, in the destination folder? I mean… if it's `current`, then it doesn't really matter… but if they want to dump the formatted files somewhere else (like we do in our tests)… I guess ignored files should still be put there… just not modified.

File: src/rebar3_formatter.erl [hunk 3]
```diff
@@ -18,27 +18,30 @@
...
-                erlang:error({modified_ast, File, NewFile})
-          end
+                changed
+            end,
+    OutputDir = maps:get(output_dir, FileOpts),
+    case maybe_save_file(OutputDir, File, Formatted) of
+        none ->
+            Result;
+        NewFile ->
+            check_quick_ast(File, NewFile, Result)
```
Comment (elbrujohalcon) [inline src/rebar3_formatter.erl:36]: If `output_dir` is `current`, i.e. if `File == NewFile`… wouldn't this check be bogus?
I mean… we will have replaced the file with the formatted version already… we won't be using the old AST for comparison.

File: test_app/after/src/brackets_ignore.erl [hunk 4]
```diff
@@ -0,0 +1,46 @@
+-module(brackets).
```
Comment (elbrujohalcon) [inline test_app/after/src/brackets_ignore.erl:1]: Should be `brackets_ignore`

File: test_app/src/brackets_ignore.erl [hunk 5]
```diff
@@ -0,0 +1,46 @@
+-module(brackets).
```
Comment (elbrujohalcon) [inline test_app/src/brackets_ignore.erl:1]: Should be `brackets_ignore`

**Pull request merged**